apps/api/src/coscrad-cli/seed-test-data-with-command.cli-command.ts

```diff
@@ -0,0 +1,147 @@
+        private readonly commandHandlerService: CommandHandlerService,
+        @Inject(ID_MANAGER_TOKEN) private readonly idManager: IIdManager,
+        @Inject(COSCRAD_LOGGER_TOKEN) private readonly logger: ICoscradLogger
+    ) {
+        super();
+    }
+
+    async run(_passedParams: string[], options: SeedTestDataWithCommandOptions): Promise<void> {
+        const { type: commandType, payloadOverrides } = options;
+
+        const fixtureFsaSearchResult = allFsas.find(({ type }) => type === commandType);
+
+        if (isUndefined(fixtureFsaSearchResult)) {
+            this.logger.log(`failed to find a fixture command of type: ${commandType}. Exiting.`);
+
+            throw new InternalError(`No fixture command of type: ${commandType} found`);
+        }
+
+        const { payload: defaultPayload } = fixtureFsaSearchResult;
+
+        const fsaToWithOverrides = {
+            type: commandType,
+            payload: {
+                ...defaultPayload,
+                ...(payloadOverrides || {}),
+            },
+        };
+
+        this.logger.log(`attempting to execute FSA: ${JSON.stringify(fsaToWithOverrides)}`);
+
+        /**
+         * TODO We need to do this conditionally, only for create commands. For
+         * this reason, it is better if this logic gets moved to a special method
+         * on the command handler or a similar service. We don't want this logic
+         * in the CLI command controller in case we need to expose it via some
+         * other transport layer.
+         */
+        const generatedId = await this.idManager.generate();
+
+        const { payload: payloadWithoutGeneratedId } = fsaToWithOverrides;
+
+        const fsaToExecute = {
+            ...fsaToWithOverrides,
+            payload: {
+                ...payloadWithoutGeneratedId,
+                aggregateCompositeIdentifier: {
+                    ...payloadWithoutGeneratedId.aggregateCompositeIdentifier,
+                    id: generatedId,
+                },
+            },
+        };
+
+        const result = await this.commandHandlerService.execute(fsaToExecute, {
+            // TODO Assign the following to a constant
+            // Do we want to allow the admin user to provide a user ID here?
+            userId: `COSCRAD_ADMIN`,
+        });
+
+        if (result === Ack) {
+            this.logger.log(`success`);
+
+            return;
+        }
+
+        const msg = `The command failed with the following errors: ${result.toString()}`;
+
+        this.logger.log(msg);
+
+        throw new InternalError(msg);
+    }
+
+    @CliCommandOption({
+        flags: '--type [commandType]',
+        description: 'the type of the command to execute',
+        required: true,
+    })
+    parseCommandType(value: string): string {
+        if (!isNonEmptyString(value)) {
+            throw new InternalError(`parameter: type must be a non-empty string`);
+        }
+
+        // TODO Should we validate the command is a known type here?
```

Comment:
Probably not a good idea since this is already the responsibility of the command handler service.

Reply:
Comment removed.